SwiftPM-PlatformExclude/FirebaseFirestoreWrap/dummy.m

```diff
@@ -16,3 +16,9 @@
 #if TARGET_OS_WATCH
 #warning "Firebase Firestore does not support watchOS"
 #endif
+
+#if (defined(TARGET_OS_VISION) && TARGET_OS_VISION) && FIREBASE_BINARY_FIRESTORE
+#warning "Firebase Firestore's binary SPM distribution does not support \
+visionOS. See workaround documented in the 10.12.0 release notes: \
+https://firebase.google.com/support/release-notes/ios#version_10120_-_july_11_2023"
```

Comment:
The part about enabling building Firestore from source is hidden below the fold. I'd suggest using this link iinstead: https://firebase.google.com/support/release-notes/ios#cloud-firestore_5

Reply:
I think the issue with `https://firebase.google.com/support/release-notes/ios#cloud-firestore_5` is the "5" on the end.  I think it maps to which referenced of "Cloud Firestore" it is on the page. The 10.12.0 release note is the 5th reference from the top, so if we use that link, I believe it will be outdated once a release goes out with a Cloud Firestore note.

Reply:
> I think the issue with `https://firebase.google.com/support/release-notes/ios#cloud-firestore_5` is the "5" on the end. I think it maps to which reference of "Cloud Firestore" it is on the page. The 10.12.0 release note is the 5th reference from the top, so if we use that link, I believe it will be outdated once a release goes out with a Cloud Firestore note.

Ah, you're right - I just tried https://firebase.google.com/support/release-notes/ios#cloud-firestore_6 to verify. Too bad.

I worry a bit that people might not see the binary build specific note about visionOS, and be side-tracked by the intro paragraph that talks about Auth and keychain. 

Maybe we should add a more detailed instruction to the build warning?

Reply:
What do you think of this? I copied and pasted the release note and adjusted it a little:

```objc
#if (defined(TARGET_OS_VISION) && TARGET_OS_VISION) && FIREBASE_BINARY_FIRESTORE
#warning "Firebase Firestore's binary SPM distribution does not support \
visionOS. To enable the source distribution, quit Xcode and open the desired \
project from the command line with the FIREBASE_SOURCE_FIRESTORE environment \
variable: `open --env FIREBASE_SOURCE_FIRESTORE /path/to/project.xcodeproj` \
To go back to using the binary distribution of Firestore, quit Xcode and open \
Xcode like normal, without the environment variable."
#endif
```

I also thought we could use Chrome's copy-link-to-highlight feature but links generated like that don't work well in Safari (and perhaps other browsers).

Reply:
<img width="580" alt="Screenshot 2024-01-16 at 11 12 24 AM" src="https://github.com/firebase/firebase-ios-sdk/assets/36927374/165b6ebf-ef96-4c98-8dd8-7d48c7d52c67">

Reply:
Looks good to me!

Reply:
Fixed in a79bae3